test/hotel_manager_test.rb

```diff
@@ -0,0 +1,347 @@
+  end 
+
+
+  describe "wave 3" do
+    describe "#find_block_reservations_by_date" do 
+      it "takes date_range & room and returns block Reservations per room" do 
+        date_range = Hotel::DateRange.new(Date.new(2020, 11, 5), Date.new(2020, 11, 11))
+        date = Date.new(2020, 11, 7)
+
+        @hotel_manager.set_block(date_range, number_of_rooms: 3)
+        @hotel_manager.set_block(date_range, number_of_rooms: 5)
+
+        @hotel_manager.make_block_reservation(date_range, number_of_rooms: 3)
+        @hotel_manager.make_block_reservation(date_range, number_of_rooms: 5)
+
+
+        blocks = @hotel_manager.find_block_reservations_by_date(date)
+
+        expect(blocks.length).must_equal 2 
+
+        expect(blocks[0].rooms.length).must_equal 3
+        expect(blocks[1].rooms.length).must_equal 5
+
+        # Refactored - checking room numbers for block
+        expect(blocks[0].rooms[0].id).must_equal 20
+        expect(blocks[0].rooms[1].id).must_equal 19
+        expect(blocks[0].rooms[2].id).must_equal 18
+
+        expect(blocks[1].rooms[0].id).must_equal 17
+        expect(blocks[1].rooms[1].id).must_equal 16
+      end 
+    end 
+
+   
+    describe "@set_block(date_range, number_of_rooms: 1)" do 
+      it "takes a date range & number of rooms and returns a Block" do 
+        block = @hotel_manager.set_block(@date_range, number_of_rooms: 4)
+
+        expect(block).must_be_kind_of Hotel::Block
+
+        expect(@hotel_manager.blocks.length).must_equal 1
+      end 
+    end 
+
+    describe "#available_blocks(date_range, number_of_rooms: number_of_rooms)" do 
+      it "returns an available block within the specific date range" do
+        @hotel_manager.set_block(@date_range, number_of_rooms: 4)
+        
+        available_blocks = @hotel_manager.available_blocks(@date_range, number_of_rooms: 4)
+
+        expect(available_blocks[0].rooms.length).must_equal 4
+      end 
+    end 
+
+
+    describe "#make_block_reservation(date_range, number_of_rooms: 2)" do 
+      it "takes a date range & number of rooms and returns a Block Reservation" do 
+        @hotel_manager.set_block(@date_range, number_of_rooms: 5)
+
+        block_reservation = @hotel_manager.make_block_reservation(@date_range, number_of_rooms: 5)
+
+        expect(block_reservation).must_be_instance_of Hotel::Reservation
+
+        expect(block_reservation.block).must_be_kind_of Hotel::Block
+
+        expect(block_reservation.block.rooms).must_be_kind_of Array
+
+        expect(block_reservation.block.rooms[0]).must_be_kind_of Hotel::Room
+      end 
+
+      it "raises an ArgumentError if the number of rooms is not between 1 and 5" do 
+
+        expect{@hotel_manager.make_block_reservation(@date_range, number_of_rooms: 6)}.must_raise ArgumentError
+
+        expect{@hotel_manager.make_block_reservation(@date_range, number_of_rooms: 0)}.must_raise ArgumentError
+      end 
+
+
+      it "raise an ArgumentError if there are no available rooms for block" do 
```

Comment:
It is also important to test that it raises in exception if there are no available rooms because the rooms are taken up by normal reservations. I believe you have implemented this functionality correctly, but it's hard to know for sure without a unit test.